auslib/web/api_validator.py

```diff
@@ -0,0 +1,30 @@
+import codecs
+import logging
+from connexion.decorators.validation import ParameterValidator
+
+log = logging.getLogger(__name__)
+
+
+class BalrogParameterValidator(ParameterValidator):
+    """
+    This class is used to keep the flask "defaults" feature behavior on spec validation level.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(BalrogParameterValidator, self).__init__(*args, **kwargs)
+
+    def validate_path_parameter(self, param, request):
+        # Keep the base class behavior.
+        val = request.path_params.get(param['name'].replace('-', '_'))
+
+        # Supporting the flask add_url_rule "defaults" parameter to the swagger path parameter.
+        if val is None and 'default' in param:
+            val = param['default']
+            request.path_params[param['name']] = val
+
+        # Encode strings in utf-8.
+        if param['type'] == 'string':
+            val = codecs.encode(val, 'utf-8')
```

Comment:
After speaking with @catlee , I don't think assuming utf-8 is a great thing to do here. We may get all sorts of different encodings, and we have no idea what will be correct. Maybe you can look into how Werkzeug or Flask does this now, and replicate that behaviour?